scripts/sign_encrypt.py

```diff
@@ -381,6 +390,109 @@ def parse(self):
+
+        img_digest = self.inf[offs:offs + digest_len]
+        print('  hash:       {}...'.format(binascii.hexlify(img_digest[:12])))
+        offs += digest_len
+        sig = self.inf[offs:offs + sig_len]
+        offs += sig_len
+
+        if img_type == SHDR_BOOTSTRAP_TA or img_type == SHDR_ENCRYPTED_TA:
+            ta_uuid = self.inf[offs:offs + UUID_SIZE]
+            print('  uuid:       {}'.format(uuid.UUID(bytes=ta_uuid)))
+            offs += UUID_SIZE
+            [ta_version] = struct.unpack('<I', self.inf[offs:offs + 4])
+            print('  ta_version: {}'.format(ta_version))
+
+            offs += 4;
+            if img_type == SHDR_ENCRYPTED_TA:
+                ehdr = self.inf[offs: offs + EHDR_SIZE]
+                offs += EHDR_SIZE
+                [enc_algo, flags, nonce_len,
+                 tag_len] = struct.unpack('<IIHH', ehdr)
+
+                enc_algo_name = 'Unkown'
+                if enc_algo in enc_tee_alg.values():
+                    enc_algo_name = value_to_key(enc_tee_algo, enc_algo)
+                print('  enc_algo:   {} ({})'.format(enc_algo, enc_algo_name))
+
+                if enc_algo not in enc_tee_alg.values():
+                    raise Exception('Unrecognized encrypt algorithm: 0x{:08x}'
+                                    .format(enc_value))
+                print('  nonce_len:  {} (bytes)'.format(nonce_len))
+                if nonce_len != NONCE_SIZE:
+                    raise Exception("Unexpected nonce len: {}"
+                                    .format(nonce_len))
+                nonce = self.inf[offs:offs + nonce_len]
+                print('  nonce:      {}...'.format(binascii.hexlify(nonce)))
+                offs += nonce_len
+
+                print('  tag_len:    {} bytes'.format(tag_len))
+                if tag_len != TAG_SIZE:
+                    raise Exception("Unexpected tag len: {}".format(tag_len))
+                tag = self.inf[-tag_len:]
+                print('  tag:        {}...'.format(binascii.hexlify(tag)))
+                ciphertext = self.inf[offs:-tag_len]
+                print('  Ciph offset:{} (0x{:x}) bytes)'.format(offs, offs))
+                print('  Ciph size:  {} (0x{:x}) (bytes)'
+                      .format(len(ciphertext), len(ciphertext)))
+                if len(ciphertext) != img_size:
+                    raise Exception("Unexpected ciphertext size: ",
+                                    "got {}, expected {}"
+                                    .format(len(ciphertext), img_size))
+            else:
+                img = self.inf[offs:]
+                print('  TA offset:  {} (0x{:x}) bytes'.format(offs, offs))
+                print('  TA size:    {} (0x{:x}) bytes'
+                      .format(len(img), len(img)))
+                if len(img) != img_size:
+                    raise Exception("Unexpected img size: got {}, expected {}"
+                                    .format(len(img), img_size))
```

Comment:
Would it make sense not to make a special case of this and print "TA offset" and "TA size" like in the else part? It is ciphertext, but it is also the TA; the fact it is encrypted doesn't matter much and it is shown by the `SHDR_ENCRYPTED_TA` type anyways.